be/src/olap/rowset/segment_v2/column_reader.cpp

```diff
@@ -375,6 +375,79 @@ Status ColumnReader::new_iterator(ColumnIterator** iterator) {
     }
 }
 
+/**
+ * @brief check if all pages encoded by local dict
+ * if the last page is encoded by dict, all pages are encoded by dict
+ * 
+ * @return true if 
+ * @return false 
+ */
+bool ColumnReader::all_pages_encoded_by_dict(ColumnIteratorOptions iter_opts, BlockCompressionCodec* codec) {
+    // go to the last page
+    RETURN_IF_ERROR(_ensure_index_loaded());
+    OrdinalPageIndexIterator last_iter(_ordinal_index.get(), _ordinal_index->num_data_pages() - 1);
+    PageHandle handle;
+    Slice page_body;
+    PageFooterPB footer;
+    iter_opts.type = DATA_PAGE;
+    ParsedPage parsed_page;
+    RETURN_IF_ERROR(read_page(iter_opts, last_iter.page(), &handle, &page_body, &footer, codec));
+    // parse data page
+    RETURN_IF_ERROR(ParsedPage::create(std::move(handle), page_body, footer.data_page_footer(),
```

Comment:
Are you sure this code could compiled? This line will return error status not boolean.